### PR TITLE
Vmclaim deletion

### PR DESCRIPTION
**Pull request by ebauman**
**What this PR does / why we need it**: Adds deletion of VMClaims and Sessions after completion

### CODE REVIEW

File: pkg/controllers/session/sessioncontroller.go [hunk 1]
```diff
@@ -145,6 +145,30 @@ func (s *SessionController) reconcileSession(ssName string) error {
...
 	timeUntilExpires := expires.Sub(now)
 
+	// clean up old (3 hours later) sessions, and only if they are finished
+	if expires.Add(time.Hour*3).Before(now) && ss.Status.Finished {
+		// first we need to delete the vmclaims
+		err := s.hfClientSet.HobbyfarmV1().VirtualMachineClaims().DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("hobbyfarm.io/session=%s", ss.Name),
+		})
+
+		if err != nil {
```
Comment (Oats87) [inline pkg/controllers/session/sessioncontroller.go:155]: Is this going to error out if there are no vmclaims found with the corresponding label?

Comment (ebauman) [reply pkg/controllers/session/sessioncontroller.go:155]: In my testing, no. I created a session with no corresponding vm claims, and the session deleted without an error.

File: pkg/controllers/session/sessioncontroller.go [hunk 2]
```diff
@@ -145,6 +145,30 @@ func (s *SessionController) reconcileSession(ssName string) error {
 
 	timeUntilExpires := expires.Sub(now)
 
+	// clean up old (3 hours later) sessions, and only if they are finished
+	if expires.Add(time.Hour*3).Before(now) && ss.Status.Finished {
```
Comment (Oats87) [inline pkg/controllers/session/sessioncontroller.go:149]: Can we define this as a constant at the top of this controller?

File: pkg/controllers/session/sessioncontroller.go [hunk 3]
```diff
@@ -145,6 +145,30 @@ func (s *SessionController) reconcileSession(ssName string) error {
...
+		glog.V(4).Infof("deleted vmclaims for old session %s", ss.Name)
+
+		// now that the vmclaims are deleted, go ahead and delete the session
+		err = s.hfClientSet.HobbyfarmV1().Sessions().Delete(ss.Name, &metav1.DeleteOptions{})
+
+		if err != nil {
+			return fmt.Errorf("error deleting session %s: %s", ss.Name, err)
+		}
+
+		glog.V(4).Infof("deleted old session %s", ss.Name)
```
Comment (Oats87) [inline pkg/controllers/session/sessioncontroller.go:168]: Can we move this to a more verbose log level, like 6?

File: pkg/controllers/session/sessioncontroller.go [hunk 4]
```diff
@@ -145,6 +145,30 @@ func (s *SessionController) reconcileSession(ssName string) error {
...
+		// first we need to delete the vmclaims
+		err := s.hfClientSet.HobbyfarmV1().VirtualMachineClaims().DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("hobbyfarm.io/session=%s", ss.Name),
+		})
+
+		if err != nil {
+			return fmt.Errorf("error deleting vmclaims with session label %s: %s", ss.Name, err)
+		}
+
+		glog.V(4).Infof("deleted vmclaims for old session %s", ss.Name)
```
Comment (Oats87) [inline pkg/controllers/session/sessioncontroller.go:159]: Can we move this to a more verbose log level, like 6?

File: pkg/sessionserver/sessionserver.go [hunk 5]
```diff
@@ -208,6 +208,9 @@ func (sss SessionServer) NewSessionFunc(w http.ResponseWriter, r *http.Request)
 	for index, vmset := range vms {
 		virtualMachineClaim := hfv1.VirtualMachineClaim{}
 		vmcId := util.GenerateResourceName("vmc", util.RandStringRunes(10), 10)
+		vmcLabels := make(map[string]string)
+		vmcLabels["hobbyfarm.io/session"] = session.Name // map vmc to session
```
Comment (Oats87) [inline pkg/sessionserver/sessionserver.go:212]: Can we define `hobbyfarm.io/session` as a constant at the top of this file?

**Review by Oats87 — CHANGES_REQUESTED** [review]

**Review by Oats87 — APPROVED** [review]

**Pull request merged**